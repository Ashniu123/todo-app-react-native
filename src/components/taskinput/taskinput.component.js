import React, { Component } from 'react';
import { View, TextInput, Alert } from 'react-native';
import { connect } from 'react-redux';

import { startAddItem } from '../../actions/tasks';
import styles from './taskinput.style';

export class Taskinput extends Component {
	constructor(props) {
		super(props);

		this.state = {
			text: ''
		}

		this.handleTextChange = this.handleTextChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}

	handleTextChange(text) {
		this.setState({ text })
	}

	handleSubmit() {
		if (this.state.text.length > 0) {
			this.props.startAddItem(this.state.text.trim())
				.then(() => {
					this.setState({
						text: ''
					});
				})
				.catch(() => {
					console.log('Error adding item (component)');
				});
		}
	}

	render() {
		return <View style={styles.textinput__view}>
			<TextInput
				value={this.state.text}
				onChangeText={this.handleTextChange}
				placeholder="What do you want to do?"
				onSubmitEditing={this.handleSubmit}
				underlineColorAndroid="#37474F"
				style={styles.textinput__input}
				selectionColor="#37474F"
			/>
		</View>
	}
}

const mapDispatchToProps = (dispatch) => ({
	startAddItem: (item) => dispatch(startAddItem(item))
});

export default connect(undefined, mapDispatchToProps)(Taskinput);