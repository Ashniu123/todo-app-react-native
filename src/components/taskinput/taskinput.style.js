import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  textinput__view: {
    backgroundColor: '#CFD8DC',
    paddingLeft: 10,
    paddingRight: 10
  },
  textinput__input: {
    color: '#455A64'
  }
});
