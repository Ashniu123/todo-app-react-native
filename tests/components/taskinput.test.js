import React from 'react';
import { shallow } from 'enzyme';

import { Taskinput } from '../../src/components/taskinput/taskinput.component';
import startAddItemMock from '../__mocks__/startAddItem.mock';

let wrapper, startAddItem;

beforeEach(() => {
  startAddItem = jest.fn(startAddItemMock);
  wrapper = shallow(<Taskinput startAddItem={startAddItem} />);
});

test('should render Taskinput component correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should add item on Textinput and carriage return if string is not empty', () => {
  const testString = 'Testing Taskinput component';
  wrapper.find('TextInput').prop('onChangeText')(testString);
  expect(wrapper.state('text')).toEqual(testString);
  wrapper.find('TextInput').prop('onSubmitEditing')();
  expect(startAddItem).toHaveBeenCalledWith(testString);
});
