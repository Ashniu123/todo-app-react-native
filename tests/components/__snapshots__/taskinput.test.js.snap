// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Taskinput component correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#CFD8DC",
      "paddingLeft": 10,
      "paddingRight": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="What do you want to do?"
    selectionColor="#37474F"
    style={
      Object {
        "color": "#455A64",
      }
    }
    underlineColorAndroid="#37474F"
    value=""
  />
</Component>
`;
